### PR TITLE
[Java] Create highlights.scm

### PR DESCRIPTION
**Pull request by PitcherTear22**
From [tree-sitter-java](https://raw.githubusercontent.com/tree-sitter/tree-sitter-java/master/queries/highlights.scm)

### CODE REVIEW

**Comment by vigoux** [comment]
Sounds great but some of the capture groups are not yet supported (like `@type.builtin`). You might want to look at already existing `highlights.scm` queries to know which modifications are needed.

**Comment by PitcherTear22** [comment]
:thumbs-up:

**Comment by PitcherTear22** [comment]
👍

**Comment by vigoux** [comment]
Hi, now that #40 is merged, you can find some indications on how to write queries and adapt Atom ones to the plugin. Feel free to ask any questions on gitter, we will be glad to answer !

**Comment by PitcherTear22** [comment]
Thanks !

**Comment by theHamsta** [comment]
@PitcherTear22 are you still intending to be working on this?

**Comment by PitcherTear22** [comment]
I don't think I can

**Comment by PitcherTear22** [comment]
> @PitcherTear22 are you still intending to be working on this?

Idk much, I try sometimes. Feel free to close this

**Comment by theHamsta** [comment]
No hurry. I just wanted to know whether you are still interested in maintaining this language. Thanks for your help!

**Comment by PitcherTear22** [comment]
> whether you are still interested in maintaining this language
Bruh it's the only language I know.

**Comment by PitcherTear22** [comment]
> whether you are still interested in maintaining this language
Bruh it's the only language I know.

**Comment by vigoux** [comment]
Would you also like to add a `locals.scm` or do we merge that now?
Thanks for the awesome work!

File: queries/java/tags.scm [hunk 1]
```diff
@@ -0,0 +1,8 @@
+(class_declaration
```
[the remark this answers is not in the document]
Comment (vigoux) [reply queries/java/tags.scm:1]: Yep, `tags.scm` don't exist for us but this file will be nearly perfect for a `locals.scm` when, the other comments will be sorted!

**Comment by vigoux** [comment]
I agree with @theHamsta comments.

One thing you can try too to lighten a bit the queries is to use the new syntax for queries:
```scheme
[
"if"
"else" 
] @conditional
```

File: queries/java/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+"default" @keyword
+"do" @keyword
+"else" @keyword
+"enum" @keyword
+"exports" @keyword
+"extends" @keyword
+"final" @keyword
+"finally" @keyword
+"for" @keyword
+"if" @keyword
```
[the remark this answers is not in the document]
Comment (PitcherTear22) [reply queries/java/highlights.scm]: Done

File: queries/java/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+"implements" @keyword
+"import" @keyword
```
[the remark this answers is not in the document]
Comment (PitcherTear22) [reply queries/java/highlights.scm]: Done

File: queries/java/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+; Literals
+
+(hex_integer_literal) @number
+(decimal_integer_literal) @number
+(octal_integer_literal) @number
+(decimal_floating_point_literal) @number
+(hex_floating_point_literal) @number
+(character_literal) @string
+(string_literal) @string
+(true) @constant.builtin
```
[the remark this answers is not in the document]
Comment (PitcherTear22) [reply queries/java/highlights.scm]: Done

File: queries/java/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+"synchronized" @keyword
+"throw" @keyword
+"throws" @keyword
+"to" @keyword
+"transient" @keyword
+"transitive" @keyword
+"try" @keyword
+"uses" @keyword
+"volatile" @keyword
+"while" @keyword
```
[the remark this answers is not in the document]
Comment (PitcherTear22) [reply queries/java/highlights.scm]: Done - also changed `for` to @repeat

File: queries/java/highlights.scm [hunk 6]
```diff
@@ -59,61 +60,79 @@
...
+"package"
+"private"
+"protected"
+"provides"
+"public"
+"requires"
+"return"
+"static"
+"strictfp"
+"switch"
```
Comment (vigoux) [inline queries/java/highlights.scm:106]: Hmm... You probably want to mark this as conditionnal instead ?

Comment (PitcherTear22) [reply queries/java/highlights.scm:106]: Lemme see java docs

Comment (PitcherTear22) [reply queries/java/highlights.scm:106]: Oh you meant only `switch`
Wait

**Comment by PitcherTear22** [comment]
> 
> 
> We are almost there, thanks for the awesome contribution !
> Do you want to be the maintainer of theses queries ? In either cases, you shoud change the `README.md` to notify that `java` is supported !

Idk if support is proper. I'll test it after 2 days when I get home. Maybe you can test it.
(It would be very easy to test because default java highlighting is uselessly limited)

File: queries/java/highlights.scm (hunk 6 above)
Comment (PitcherTear22) [reply queries/java/highlights.scm]: Done

Comment (vigoux) [reply queries/java/highlights.scm]: And maybe `case` and such

**Comment by vigoux** [comment]
Nothing is 100% supported on the first try, the it will be a long term effort to have a proper support for everytying. The queries you have here are the first attempt !

File: queries/java/highlights.scm (hunk 6 above)
Comment (PitcherTear22) [reply queries/java/highlights.scm]: Oh shoot, sorry! wait

Comment (PitcherTear22) [reply queries/java/highlights.scm]: Done

File: queries/java/highlights.scm [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+(string_literal) @string
+(null_literal) @constant.builtin
+
+(comment) @comment
+
+[
+"true"
+"false"
+] @boolean
+
```
Comment (theHamsta) [inline queries/java/highlights.scm:71]: Does Neovim accept your queries?

For me this has to be changed to
```
[
 (true)
 (false)
] @boolean
```

Comment (PitcherTear22) [reply queries/java/highlights.scm:71]: I haven't tested it because I only have a windows laptop rn. I'll test it tomorrow, thanks

Comment (PitcherTear22) [reply queries/java/highlights.scm:71]: I changed it anyway. Since you seem to have tested it, does highlighting work?

**Pull request closed without merging**